app/src/main/java/protect/card_locker/ScanActivity.java

```diff
@@ -272,9 +269,15 @@ private void returnResult(String barcodeContents, String barcodeFormat) {
     private void handleActivityResult(int requestCode, int resultCode, Intent intent) {
         super.onActivityResult(requestCode, resultCode, intent);
 
+        if (resultCode == Activity.RESULT_CANCELED) {
```

Comment:
With the second fix below, this is technically superfluous as `parseSetBarcodeActivityResult()` will return an empty `BarcodeValues` if `resultCode == Activity.RESULT_CANCELED`. But keeping it to be more explicit about the control flow on cancel also has benefits IMO. Up to you.

Reply:
> technically superfluous

Because `parseSetBarcodeActivityResult()` starts with:

```java
        if (resultCode != Activity.RESULT_OK) {
            return new BarcodeValues(null, null);
        }
```

I just overlooked that it would also be empty if it failed to get a barcode from the image. Was mostly looking at ways the flow could be cancelled in a way that would fail to set the scanner to active again.